File: removeKFromList/code.cs
ListNode<int> removeKFromList(ListNode<int> l, int k) {
    if(l == null)
        return null;
    
    if(l.value == k){
        var m = l.next;
        l = null;
        return removeKFromList(m, k);
    }
    
    if(l.next == null){
        return l;
    }
    else{
        if(l.next.value == k){
            if(l.next.next == null){
                l.next = null;
                return l;
            }
            else{
                l.next = l.next.next;
                return removeKFromList(l, k);
            }
        }
        else{
            return removeKFromList(l.next, k);
        }
    }
    return l;
}

File: mergeTwoLinkedLists/code.cpp
// Definition for singly-linked list:
// template<typename T>
// struct ListNode {
//   ListNode(const T &v) : value(v), next(nullptr) {}
//   T value;
//   ListNode *next;
// };
//

ListNode<int> * mergeTwoLinkedLists(ListNode<int> * l1, ListNode<int> * l2) {
        ListNode <int >result (0), *p = &result;
        while (l1 || l2){
                if (!l1 || l2 && l1->value > l2->value){
                        p = p->next = l2;
                        l2 = l2->next;
                }else{
                        p = p->next = l1;
                        l1 = l1->next;
                }
        }
        return result.next;
}
